### DISCUSSION TITLE
Null interception in PrettyPrinter

### DISCUSSION THREAD

**Discussion by kindermannhubert** [waf/CSharpRepl#88] [General]
@waf what is the reasoning behind this?
![image](https://user-images.githubusercontent.com/11704036/158054262-d1dd9d59-3d90-47cb-a79d-4190d7e12a9b.png)

csi.exe and Immediate window in VS also outputs `null` text and I would probably prefer it too.

This could surely be configurable but I'm not sure if it's necessary.

**Comment by waf** [comment]
@kindermannhubert the reasoning is to avoid the following behavior, where assignments (e.g. `var x = 5;`) return a null object from the Roslyn APIs:

![image](https://user-images.githubusercontent.com/97195/158058715-dba4d07e-e1b8-486d-9c9e-84ae8cb1be61.png)

I agree that the behavior in csi.exe is better here, where it doesn't display null for assignments, but does display in cases where the object being evaluated is null -- I believe back when this was written (a while ago now!) I didn't see a way of distinguishing the two cases in the script evaluation API.

![image](https://user-images.githubusercontent.com/97195/158058830-288f9af8-863b-42b9-8d72-dd1d65d64cb8.png)

**Comment by kindermannhubert** [comment]
Indeed Roslyn provides no information on whether the input is void-returning/no-output or value-returning. I've prepared a custom analysis of this in #187.
